types/codec.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+package cosmos_sdk;
+option go_package = "types";
+
+import "gogoproto/gogo.proto";
+
+message Coin {
+    option (gogoproto.stringer) = false;
+    string denom = 1;
+    string amount = 2 [(gogoproto.customtype) = "Int", (gogoproto.nullable) = false];
```

Comment:
`string` seems like the right cross-platform type for `sdk.Int`. `gogoproto.customtype` currently only works for `bytes`. We could make a PR to gogoproto to support string or think of another approach inside the SDK to convert to `sdk.Int`.